dashboard/src/components/CommitNavigationGraph/CommitNavigationGraph.tsx

```diff
@@ -2,42 +2,46 @@ import { useIntl } from 'react-intl';
 
 import { memo, useMemo } from 'react';
 
-import { useNavigate, useParams, useSearch } from '@tanstack/react-router';
-
 import { z } from 'zod';
 
 import { Colors } from '@/components/StatusChart/StatusCharts';
 import { LineChart } from '@/components/LineChart';
 import BaseCard from '@/components/Cards/BaseCard';
-import { useTreeCommitHistory } from '@/api/TreeDetails';
 import type { TLineChartProps } from '@/components/LineChart/LineChart';
 import QuerySwitcher from '@/components/QuerySwitcher/QuerySwitcher';
 import type { MessagesKey } from '@/locales/messages';
 import { formatDate } from '@/utils/utils';
 import { mapFilterToReq } from '@/pages/TreeDetails/TreeDetailsFilter';
+import type { TFilter } from '@/types/tree/TreeDetails';
+import { useCommitHistory } from '@/api/commitHistory';
 
 const graphDisplaySize = 7;
 
-const CommitNavigationGraph = (): JSX.Element => {
+interface ICommitNavigationGraph {
+  origin: string;
+  currentPageTab: string;
+  diffFilter: TFilter;
+  gitUrl: string;
+  gitBranch: string;
+  headCommitHash: string;
```

Comment:
Make things that can be undefined optional here

Reply:
nice